### PR TITLE
feat: service-user bo function 

### PR DESCRIPTION
**Pull request by Yakitrak**
## Describe your changes

Ticket: https://pins-ds.atlassian.net/browse/ASB-2011

This PR creates a back office subscriber function to consume service-user messages and save it to our database. It also includes a prisma schema update to match the [message schema](https://github.com/Planning-Inspectorate/back-office/blob/497d54c52e4e84bd329b2fb469e154726cb22e7d/apps/api/src/message-schemas/events/service-user.schema.json).

## Useful information to review or test

1. Follow the readme on the back-office subscribers function to set it up
2. Run azurite `npx azurite` and this function `npm run start:dev nsip-service-user`
3. Make a POST request to `http://localhost:7071/admin/functions/nsip-service-user` with this body:
```
{
    "input": "{\"serviceUserType\":\"Applicant\",\"id\":\"1234\",\"firstName\":\"John\",\"lastName\":\"Smith\",\"organisation\":\"ACME\",\"organisationType\":\"Sole Trader\",\"caseReference\":\"B123456\",\"emailAddress\":\"test@test.com\",\"webAddress\":\"www.test.com\",\"telephoneNumber\":\"0123456789\"}"
}
```

Note, the `serviceUserType` above can be changed to `Applicant`, `Representative `, and `Represented` which save different fields into the database.

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [x] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: packages/back-office-subscribers/nsip-service-user/index.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+const mapMessageToServiceUser = (message) => {
+	const serviceUserType = message.serviceUserType;
+	if (!['Representative', 'Represented', 'Applicant'].includes(serviceUserType)) {
+		throw new Error(`Invalid serviceUserType: ${serviceUserType}`);
+	}
+	if (serviceUserType === 'Representative' || serviceUserType === 'Represented') {
+		return {
+			serviceUserId: message.id,
+			firstName: message?.firstName,
```
Comment (JamesP-KC) [inline packages/back-office-subscribers/nsip-service-user/index.js:52]:
```suggestion
			firstName: message.firstName,
```

is the `?` needed (on this line and all below)? I think `message` should always be defined

**Review by JamesP-KC — APPROVED** [review]
Looks good to me 👍

One minor comment

File: packages/back-office-subscribers/nsip-service-user/index.js (hunk 1 above)
Comment (Yakitrak) [reply packages/back-office-subscribers/nsip-service-user/index.js]: Good spot thanks - not sure why I had that

**Pull request merged**